src/services/directory.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { backendFetchJson, getRequestParamFromList } from './utils';
+import { UUID } from 'crypto';
+import { ElementAttributes } from '../utils/ElementAttributes';
+
+const PREFIX_DIRECTORY_SERVER_QUERIES =
+    import.meta.env.VITE_API_GATEWAY + '/directory';
+
+export function fetchRootFolders(
+    types: string[]
+): Promise<ElementAttributes[]> {
+    console.info('Fetching Root Directories');
+
+    // Add params to Url
+    const urlSearchParams = getRequestParamFromList(
+        types,
+        'elementTypes'
+    ).toString();
+    const fetchRootFoldersUrl = `${PREFIX_DIRECTORY_SERVER_QUERIES}/v1/root-directories?${urlSearchParams}`;
+    return backendFetchJson(fetchRootFoldersUrl, {
+        method: 'get',
+        headers: { 'Content-Type': 'application/json' },
+    });
+}
+
+export function fetchDirectoryContent(
+    directoryUuid: UUID,
+    types?: string[]
+): Promise<ElementAttributes[]> {
+    console.info("Fetching Folder content '%s'", directoryUuid);
+
+    // Add params to Url
+    const urlSearchParams = getRequestParamFromList(
+        types,
+        'elementTypes'
+    ).toString();
+
+    let fetchDirectoryContentUrl = `${PREFIX_DIRECTORY_SERVER_QUERIES}/v1/directories/${directoryUuid}/elements`;
+    if (urlSearchParams.toString()) {
+        fetchDirectoryContentUrl += `?${urlSearchParams}`;
+    }
```

Comment:
```suggestion
    const fetchDirectoryContentUrl = `${PREFIX_DIRECTORY_SERVER_QUERIES}/v1/directories/${directoryUuid}/elements${urlSearchParams ? '?' + urlSearchParams : ''}`;
```

Reply:
An empty URLSearchParams still returns true whereas an empty string from an empty URLSearchParams is falsy so we ought to test on `urlSearchParams.toString()`

Reply:
My bad